### PR TITLE
tev: Use primary color for previous source in first stage

### PR DESCRIPTION
**Pull request by GPUCode**
Fixes an edge case when games use GPU_PREVIOUS source on the first tev stage. Hardware tests have shown that in this case the primary color is used, instead of zero which citra currently implements. The test attached configures a triangle with blue primary color and sets the first tev stage to GPU_PREVIOUS. Pressing X changes the color. This behaviour also matches the OpenGL ES 1.1 [specification](https://registry.khronos.org/OpenGL-Refpages/es1.1/xhtml/glTexEnv.xml)

The simple program attached also tests of GPU_PREVIOUS_BUFFER. That one seems to always return zero when used on the first stage so the current citra behaviour is correct on that one. Feedback regarding the test results would be appreciated

3DS | Citra (master) | Citra (PR)
-|-|-
3ds | citra | citra_new

Fixes https://github.com/citra-emu/citra/issues/6101
Kirby's Blowout Blast_25 08 23_13 55 45 069

[tev_test.zip](https://github.com/citra-emu/citra/files/12438404/tev_test.zip)

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6921)

### CODE REVIEW

File: src/video_core/renderer_opengl/gl_shader_gen.cpp [hunk 1]
```diff
@@ -1479,7 +1479,7 @@ vec4 secondary_fragment_color = vec4(0.0);
 
     out += "vec4 combiner_buffer = vec4(0.0);\n"
            "vec4 next_combiner_buffer = tev_combiner_buffer_color;\n"
-           "vec4 last_tex_env_out = vec4(0.0);\n";
+           "vec4 last_tex_env_out = primary_color;\n";
```
Comment (Steveice10) [inline src/video_core/renderer_opengl/gl_shader_gen.cpp:1482]: Should probably be ```rounded_primary_color``` for precision. Not sure if this applies to any other backend.

**Review by Steveice10 — APPROVED** [review]

**Comment by windwakr** [comment]
This changes the color selection boxes in Fantasy Life from black to white. Still incorrect, though.

Nightly 1978
nightly1978

This PR
pr

**Comment by GPUCode** [comment]
The game also uses GPU_PREVIOUS in the first stage for drawing the boxes, which explains why it was affected. It appears the color for each box is set to GPU_CONSTANT but the game doesn't configure TEV to use it?

**Comment by FearlessTobi** [comment]
Do you think this could also fix https://github.com/citra-emu/citra/issues/2236?

**Review by FearlessTobi — APPROVED** [review]

**Comment by Steveice10** [comment]
> Do you think this could also fix #2236?

Doesn't seem like it.
